rclcpp_lifecycle/src/lifecycle_node_interface_impl.hpp

```diff
@@ -265,6 +265,8 @@ class LifecycleNode::LifecycleNodeInterfaceImpl
       throw std::runtime_error(
               "Can't get available states. State machine is not initialized.");
     }
+
+    resp->available_states.reserve(state_machine_.transition_map.states_size);
     for (uint8_t i = 0; i < state_machine_.transition_map.states_size; ++i) {
       lifecycle_msgs::msg::State state;
       state.id = static_cast<uint8_t>(state_machine_.transition_map.states[i].id);
```

Comment:
This sequence currently does the construction of the `state` object, sets the fields, and then `push_back` onto the vector (invoking a copy, if I understand properly).

What if instead we did an `emplace_back` of the default-constructed object (avoiding the copy), and then set the fields?  Do you think that would be any faster?  Same question below.

Reply:
That's a good point. I think for the service callback loops it makes sense to use resize to allocate and construct in one go. Then each field can be assigned to. I think emplace_back would have similar performance to resize (but better than reserve).

For the service callbacks though, the performance differential is minimal relative to a service call in general though. So the benchmarks don't really show an improvement for this service call.